.github/workflows/release.yaml

```diff
@@ -0,0 +1,162 @@
+
+    - name: Download cross from GitHub releases
+      uses: giantswarm/install-binary-action@v1.0.0
+      if: ${{ matrix.job.use-cross }}
+      with:
+        binary: "cross"
+        version: "v0.2.1"
+        download_url: "https://github.com/rust-embedded/cross/releases/download/${version}/cross-${version}-x86_64-unknown-linux-gnu.tar.gz"
+        tarball_binary_path: "${binary}"
+        smoke_test: "${binary} --version"
+
+    - name: Build
+      run: |
+        if [ "${{ matrix.job.use-cross }}" == "true" ]; then
+          cross build --release --target=${{ matrix.job.target }}
+        else
+          cargo build --release --target=${{ matrix.job.target }}
+        fi
+
+    - name: Rename lib to the final name
+      id: rename
+      run: |
+        LIB_PREFIX="lib"
+        case ${{ matrix.job.target }} in
+          *-pc-windows-*) LIB_PREFIX="" ;;
+        esac;
+
+        # Figure out suffix of lib
+        # See: https://doc.rust-lang.org/reference/linkage.html
+        LIB_SUFFIX=".so"
+        case ${{ matrix.job.target }} in
+          *-apple-darwin) LIB_SUFFIX=".dylib" ;;
+          *-pc-windows-*) LIB_SUFFIX=".dll" ;;
+        esac;
+
+        CICD_INTERMEDIATES_DIR=$(mktemp -d) 
+
+        # Setup paths
+        LIB_DIR="${CICD_INTERMEDIATES_DIR}/released-lib"
+        mkdir -p "${LIB_DIR}"
+        LIB_NAME="${LIB_PREFIX}${{ env.PROJECT_NAME }}${LIB_SUFFIX}"
+        LIB_PATH="${LIB_DIR}/${LIB_NAME}"
+
+        # Copy the release build lib to the result location
+        cp "target/${{ matrix.job.target }}/release/${LIB_NAME}" "${LIB_DIR}"
+
+        # Final paths
+        # In the end we use ".so" for MacOS in the final build
+        # See: https://www.erlang.org/doc/man/erlang.html#load_nif-2
+        LIB_FINAL_SUFFIX="${LIB_SUFFIX}"
+        case ${{ matrix.job.target }} in
+          *-apple-darwin) LIB_FINAL_SUFFIX=".so" ;;
+        esac;
+
+        LIB_FINAL_NAME="${LIB_PREFIX}${PROJECT_NAME}-v${PROJECT_VERSION}-nif-${RUSTLER_NIF_VERSION}-${{ matrix.job.target }}${LIB_FINAL_SUFFIX}"
+
+        # Copy lib to final name on this directory
+        cp "${LIB_PATH}" "${LIB_FINAL_NAME}"
+
+        tar -cvzf "${LIB_FINAL_NAME}.tar.gz" "${LIB_FINAL_NAME}"
+
+        # Passes the path relative to the root of the project.
+        LIB_FINAL_PATH="${NIF_DIRECTORY}/${LIB_FINAL_NAME}.tar.gz"
+
+        # Let subsequent steps know where to find the lib
+        echo ::set-output name=LIB_FINAL_PATH::${LIB_FINAL_PATH}
+        echo ::set-output name=LIB_FINAL_NAME::${LIB_FINAL_NAME}.tar.gz
+
+    - name: "Artifact upload"
+      uses: actions/upload-artifact@v2
+      with:
+        name: ${{ steps.rename.outputs.LIB_FINAL_NAME }}
+        path: ${{ steps.rename.outputs.LIB_FINAL_PATH }}
+
+    - name: Publish archives and packages
+      uses: softprops/action-gh-release@v1
+      with:
+        files: |
+          ${{ steps.rename.outputs.LIB_FINAL_PATH }}
+      if: startsWith(github.ref, 'refs/tags/')
```

Comment:
Ah, do I read this correctly? This creates a GH release automatically is a commit on `main` is tagged, right?

Reply:
The rules are below:
1. Always run on main branch
2. Run if "native" path changed
3. Tags will always run


 It creates a GH release only when a new tag is pushed, it only runs building test on `main` branch pushes or native folder changes.

| branch | building test | creating a release |
|---|---|---|
| `master` | ✅ | ❌ |
| `feat-branch` | ❌ | ❌ |
| `feat-branch-with-native-changes` | ✅ | ❌ |
| `v1.0.1` | ✅ | ✅ |

Reply:
> Ah, do I read this correctly? This creates a GH release automatically is a commit on `main` is tagged, right?

It only creates a GH release automatically when a tag pushed.

Reply:
Thanks for the explanation!